mxcubeweb/core/server/resource_handler.py

```diff
@@ -346,7 +346,7 @@ def _validate_param_data(
                 # We only handle a single argument when using Pydantic models
                 # We handle complex structures by checking if param_data
                 # contains a value or a dictionary.
-                if isinstance(param_data[param_name], dict):
+                if isinstance(param_data.get(param_name), dict):
                     return param_type.parse_obj(param_data[param_name])
```

Comment:
Assign to a variable instead of repeating operations:
```suggestion
                some_fancy_name = param_data.get(param_name)
                if isinstance(some_fancy_name, dict):
                    return param_type.parse_obj(some_fancy_name)
```